test/com/google/enterprise/adaptor/filenet/PermissionsTest.java

```diff
@@ -691,4 +694,275 @@ public void testMarking_UserNotMatchingAnyAces() {
+    assertTrue(acl.getDenyGroups().toString(), acl.getDenyGroups().isEmpty());
+  }
+
+  @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+  @Test
+  public void testMarkingAcl_noUseMarking_allowConstraintMask()
+      throws Exception {
+    Permissions testPerms = new Permissions(
+        TestObjectFactory.getPermissions(PermissionSource.MARKING));
+    Permissions.Acl acl = testPerms.getMarkingAcl(~VIEW_ACCESS_RIGHTS);
+
+    assertEquals(ImmutableSet.<String>of(AUTHENTICATED_USERS),
+        acl.getAllowGroups(PermissionSource.MARKING));
+    assertEquals(ImmutableSet.<String>of(AUTHENTICATED_USERS),
+        acl.getAllowGroups());
+    assertTrue(acl.getAllowUsers().toString(), acl.getAllowUsers().isEmpty());
+    assertTrue(acl.getDenyUsers().toString(), acl.getDenyUsers().isEmpty());
+    assertTrue(acl.getDenyGroups().toString(), acl.getDenyGroups().isEmpty());
+  }
+
+  @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+  @Test
+  public void testMarkingAcl_noUseMarking_denyConstraintMask()
+      throws Exception {
+    Permissions testPerms = new Permissions(
+        TestObjectFactory.getPermissions(PermissionSource.MARKING));
+    Permissions.Acl acl = testPerms.getMarkingAcl(VIEW_ACCESS_RIGHTS);
+
+    assertTrue(acl.getAllowUsers().toString(), acl.getAllowUsers().isEmpty());
+    assertTrue(acl.getAllowGroups().toString(), acl.getAllowGroups().isEmpty());
+    assertTrue(acl.getDenyUsers().toString(), acl.getDenyUsers().isEmpty());
+    assertTrue(acl.getDenyGroups().toString(), acl.getDenyGroups().isEmpty());
+  }
+
+  @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+  @Test
+  public void testMarkingAcl_noUseMarking_partialConstraintMask()
+      throws Exception {
+    Permissions testPerms = new Permissions(
+        TestObjectFactory.getPermissions(PermissionSource.MARKING));
+    Permissions.Acl acl = testPerms.getMarkingAcl(~AccessRight.READ_AS_INT);
+
+    assertTrue(acl.getAllowUsers().toString(), acl.getAllowUsers().isEmpty());
+    assertTrue(acl.getAllowGroups().toString(), acl.getAllowGroups().isEmpty());
+    assertTrue(acl.getDenyUsers().toString(), acl.getDenyUsers().isEmpty());
+    assertTrue(acl.getDenyGroups().toString(), acl.getDenyGroups().isEmpty());
+  }
+
+
+  @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+  @Test
+  public void testMarkingAcl_allUseMarking_allowConstraintMask()
+      throws Exception {
+    Permissions testPerms = new Permissions(
+        TestObjectFactory.newPermissionList(
+            TestObjectFactory.generatePermissions(2, 2, 1, 2,
+                AccessRight.USE_MARKING_AS_INT, 0, PermissionSource.MARKING)));
+    Permissions.Acl acl = testPerms.getMarkingAcl(~VIEW_ACCESS_RIGHTS);
+
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_allow_user_0", "MARKING_allow_user_1"),
+        acl.getAllowUsers());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_allow_group_0", AUTHENTICATED_USERS),
```

Comment:
The use of 2, 2, 1, 2, in a different order than these assertions, for the purpose, I'm guessing, of not wrapping this line seems borderline obfuscated.